openedx/core/djangoapps/content/search/api.py

```diff
@@ -415,7 +421,35 @@ def add_with_children(block):
                 num_contexts_done += 1
                 num_blocks_done += len(docs)
 
-    status_cb(f"Done! {num_blocks_done} blocks indexed across {num_contexts_done} courses and libraries.")
+        ############## Collections ##############
```

Comment:
This is starting to get really big.

We have a lot of local variables here, but could you try to create some inner functions like `config_index()`, `index_libraries()`, `index_courses()`, and `index_collections()`?

Reply:
@rpenido Not sure if it improves readability but converted some parts to inner functions.